src/UXClient/Components/CategoricalPlot/CategoricalPlot.ts

```diff
@@ -0,0 +1,189 @@
+import * as d3 from 'd3';
+import { Plot } from '../../Interfaces/Plot';
+import { Utils, NONNUMERICTOPMARGIN, DataTypes, LINECHARTTOPPADDING } from '../../Utils';
+
+const TOPMARGIN = 4;
+
+class CategoricalPlot extends Plot {
+    private defs;
+    private hoverRect;
+    private chartGroup;
+    private categoricalMouseover;
+    private categoricalMouseout;
+    private splitBysGroup;
+
+    constructor (svgSelection) {
+        super(svgSelection);
+        this.plotDataType = DataTypes.Categorical;
+    }
+
+    private onMouseover (d, rectWidth) {
+        let visibleMeasures = this.getVisibleMeasures(d.measures);
+
+        this.hoverRect.attr('visibility', 'visible')
+            .attr('x', () => {
+                return this.x(new Date(d.dateTime))
+            })
+            .attr('width', rectWidth)
+            .attr('height', this.chartHeight + 1 - LINECHARTTOPPADDING)
+            .attr('fill', () => {
+                return visibleMeasures.length === 1 ? this.getColorForValue(visibleMeasures[0]) : 'none'
+            });
+    }
+
+    private onMouseout () {
+        this.hoverRect.attr('visibility', 'hidden');
+        this.categoricalMouseout();
+    }
+
+    private createHoverRect () {
+        if (!this.hoverRect) {
+            this.hoverRect = this.chartGroup.append('rect')
+                .attr('class', 'tsi-categoricalHoverRect')
+                .attr('y', LINECHARTTOPPADDING)
+                .attr('height', this.chartHeight + 1)
+        } else {
+            this.hoverRect.raise();
+        }
+    }
+
+    private getSeriesEndDate () {
+        if (this.chartDataOptions.searchSpan) {
+            return new Date(this.chartDataOptions.searchSpan.to);
+        }
+        return new Date(this.chartComponentData.toMillis);
+    }
+
+    private getBucketEndDate (d, i) {
+        let data = this.chartComponentData.timeArrays[d.aggregateKey][d.splitBy];
+        if (i + 1 < data.length) {
+            return data[i+1].dateTime; 
+        } else {
+            let shouldRoundEnd = Utils.safeNotNullOrUndefined(() => this.chartDataOptions.searchSpan) && Utils.safeNotNullOrUndefined(() => this.chartDataOptions.searchSpan.bucketSize);
+            return shouldRoundEnd ? Utils.roundToMillis(this.getSeriesEndDate().valueOf(), Utils.parseTimeInput(this.chartDataOptions.searchSpan.bucketSize)) : this.getSeriesEndDate();        
+        }
+    }
+
+    public render (chartOptions, visibleAggI, agg, aggVisible: boolean, aggregateGroup, chartComponentData, yExtent,  
```

Comment:
consider using an object for these parameters and spreading it to "this" for the 1.3.0 release